c_src/membrane_opus_plugin/decoder.c

```diff
@@ -42,22 +42,26 @@ UNIFEX_TERM decode_packet(UnifexEnv *env, State *state,
 
   unsigned output_size =
       samples_per_channel * state->channels * sizeof(opus_int16);
-  UnifexPayload *out_payload =
-      unifex_payload_alloc(env, UNIFEX_PAYLOAD_BINARY, output_size);
+
+  UnifexPayload out_payload;
+  unifex_payload_alloc(env, UNIFEX_PAYLOAD_BINARY, output_size, &out_payload);
 
   int decoded_samples_per_channel =
       opus_decode(state->decoder, in_payload->data, in_payload->size,
-                  (opus_int16 *)out_payload->data, out_payload->size, 0);
+                  (opus_int16 *)out_payload.data, out_payload.size, 0);
   if (decoded_samples_per_channel < 0) {
     error = (char *)opus_strerror(decoded_samples_per_channel);
-    goto decode_packet_error;
+    goto decode_packet_error_and_cleanup;
   }
   if (decoded_samples_per_channel != samples_per_channel) {
     error = "invalid decoded output size";
-    goto decode_packet_error;
+    goto decode_packet_error_and_cleanup;
   }
 
-  return decode_packet_result(env, out_payload);
+  return decode_packet_result(env, &out_payload);
+
+decode_packet_error_and_cleanup:
+  unifex_payload_release(&out_payload);
 decode_packet_error:
   return unifex_raise(env, error);
```

Comment:
Maybe that's a safer approach?

```suggestion
decode_packet_error:
  if (out_payload) {
    unifex_payload_release(&out_payload);
  }
  return unifex_raise(env, error);
```

Though remember to set `out_payload = {0};`

Reply:
We probably can't use struct value in if statement. This gives me 
```
error: used struct type value where scalar is required
   64 |   if (out_payload) {
```